### ISSUE TITLE
Receipt formatter doesn't format 'from' and 'to' fields to checksum addresses

### ISSUE THREAD

**Issue by ignition42** [ethereum/web3.py#1547]
### What was wrong?

Since `w3.eth.getTransaction()` returns the `from` and `to` as checksum addresses I was expecting the same behaviour for the receipt methods like `w3.eth.waitForTransactionReceipt()` or `w3.eth.getTransactionReceipt()`.

Basically the [receipt formatter](https://github.com/ethereum/web3.py/blob/v5.4.0/web3/_utils/method_formatters.py#L188) used [pythonic middleware](https://github.com/ethereum/web3.py/blob/v5.4.0/web3/middleware/pythonic.py) does not format `from` and `to` addresses to checksum as it does with [transaction formatter](https://github.com/ethereum/web3.py/blob/v5.4.0/web3/_utils/method_formatters.py#L139). So when I call `w3.eth.waitForTransactionReceipt()` or `w3.eth.getTransactionReceipt()`, `from`
 and `to` fields non checksumed addresses in lowercase. 

### How can it be fixed?

Basically add them here: https://github.com/ethereum/web3.py/blob/v5.4.0/web3/_utils/method_formatters.py#L174

As they are here: https://github.com/ethereum/web3.py/blob/v5.4.0/web3/_utils/method_formatters.py#L119

**Comment by rajeev29five** [comment]
I would like to work on this.

**Comment by ashish10677** [comment]
@rajeev29five are you trying to fix this issue or can I also try?

**Comment by rajeev29five** [comment]
@ashish10677 I have already fixed the issue, just need to create a PR.
